addons/website_livechat/static/src/models/thread_view/thread_view.js

```diff
@@ -0,0 +1,31 @@
+/** @odoo-module **/
+
+import {
+    registerFieldPatchModel,
+    registerInstancePatchModel,
+} from '@mail/model/model_core';
+import { attr } from '@mail/model/model_field';
+
+registerInstancePatchModel('mail.thread_view', 'website_livechat/static/src/models/thread_view/thread_view.js', {
+
+    //----------------------------------------------------------------------
+    // Private
+    //----------------------------------------------------------------------
+
+    /**
+     * @private
+     * @returns {boolean}
+     */
+    _computeHasVisitorBanner() {
+        return this.thread && this.thread.visitor && this.threadViewer && this.threadViewer.discuss;
```

Comment:
I got traceback when testing this and changing channels in discuss.

Issue is that this is lacking cast to boolean. When thread becomes empty the compute returns undefined which does nothing.

```
UncaughtPromiseError > TypeError
Uncaught Promise > scope.threadView.thread.visitor is undefined
TypeError: scope.threadView.thread.visitor is undefined
    anonymous@https://14281968-15-0-all.runbot40.odoo.com/web/assets/1644-d011ec2/web.assets_common.min.js line 1016 > Function:50:29
    render@https://14281968-15-0-all.runbot40.odoo.com/web/assets/1644-d011ec2/web.assets_common.min.js:1005:20
    __render@https://14281968-15-0-all.runbot40.odoo.com/web/assets/1644-d011ec2/web.assets_common.min.js:1334:33
    useModels/component.__render@https://14281968-15-0-all.runbot40.odoo.com/web/assets/1671-b518a90/web.assets_backend.min.js:6221:10
    render/<@https://14281968-15-0-all.runbot40.odoo.com/web/assets/1644-d011ec2/web.assets_common.min.js:1298:6
    
```